### DISCUSSION TITLE
Getting Provider of the wallet to trigger the contract functions

### DISCUSSION THREAD

**Discussion by Abubakar672** [WalletConnect/walletconnect-monorepo#1138] [Developer Support] [answered]
Below I have provided the code that is working up till now to get a connection from the metamask to my sample application, Now I am struggling to get the provider of the wallet to trigger and use the smart contract functions. 

`
import WalletConnectProvider, {

  useWalletConnect,

} from "@walletconnect/react-native-dapp";

import AsyncStorage from "@react-native-async-storage/async-storage";

function SampleWCConnect() {

  const connector = useWalletConnect();

  const connectWallet = React.useCallback(() => {

    return connector.connect();

   }, [connector]);

  //const connectWallet =async()=>{

    //await provider.enable();

  //}

  const killSession = React.useCallback(async () => {

    return connector.killSession();

  }, [connector]);

  async function getETH() {

    console.log("Hello getETH");

    Linking.openURL("[https://metamask.app.link/dapp/sampledapp");](https://metamask.app.link/dapp/sampledapp%22);)

  }

return (

    <View style={styles.container}>

      {!connector.connected ? (

        <TouchableOpacity

          style={styles.buttonStyle}

          activeOpacity={0.5}

          onPress={connectWallet}

        >

          <Text style={styles.buttonTextStyle}>Connect a Wallet</Text>

        </TouchableOpacity>

      ) : null}

      {!!connector.connected ? (

        <>

          <Text>Address: {shortenAddress(connector.accounts[0])}</Text>

          {/* {loadContract(connector)} */}

          <Line />

          <TouchableOpacity

            style={styles.buttonStyle}

            activeOpacity={0.5}

            onPress={killSession}

          >

            <Text style={styles.buttonTextStyle}>Log Out</Text>

          </TouchableOpacity>

        </>

      ) : null}

    </View>

  );

}

export default function WallectConnectMain() {

  return (

    <WalletConnectProvider

      storageOptions={{ asyncStorage: AsyncStorage }}

      redirectUrl="myapp://"

    >

      <SampleWCConnect />

    </WalletConnectProvider>

  );

}
`

https://user-images.githubusercontent.com/50159697/172129916-47abf099-6ed9-45ff-9f48-4d73f16a37a2.mp4

**Comment by Abubakar672** [comment]
@hackyguru  can you please help me in this i want to know now how can i get the provider so i can call the contract functions.

**Comment by hackyguru** [comment, marked as answer]
This could be useful,

**Enabling the provider to pop-up QR modal**
```
await provider.enable();
```

**Disconnecting the provider**
```
await provider.disconnect()
```

**Sending JSON RPC requests**
```
const result = await provider.request(payload: RequestArguments);
```

You can connect to a contract and use its functions using Ethers/Web3.js.

**Reply by Abubakar672** [reply]
Thanks for the reply @hackyguru i will try this.
